src/Idephix/Console/Command.php

```diff
@@ -1,61 +1,126 @@
-        $this->idxTask = $code;
+        $command = new static($name);
+        $command->idx = $idx;
+
+        $command->assertCallable($code);
+        $command->idxTaskCode = $code;
 
         $reflector = new \ReflectionFunction($code);
         $parser = new DocBlockParser($reflector->getDocComment());
-        $this->setDescription($parser->getDescription());
+        $command->setDescription($parser->getDescription());
 
         foreach ($reflector->getParameters() as $parameter) {
             $description = $parser->getParamDescription($parameter->getName());
 
             if ($parameter->getName() !== 'idx') {
-                $this->addParameter($parameter, $description);
+                $command->addParameter($parameter, $description);
             }
         }
 
-        return $this;
+        return $command;
+    }
+
+    /**
+     * @param Task $task
+     * @return Command
+     */
+    public static function fromTask(Task $task, IdephixInterface $idx)
+    {
+        $command = new static($task->name());
+        $command->idx = $idx;
+
+        $command->setDescription($task->description());
+
+        /** @var Parameter $parameter */
+        foreach ($task->parameters() as $parameter) {
+            if (!$parameter->isOptional()) {
+                $command->addArgument($parameter->name(), InputArgument::REQUIRED, $parameter->description());
+                continue;
+            }
+
+            if ($parameter->isFlagOption()) {
+                $command->addOption($parameter->name(), null, InputOption::VALUE_NONE, $parameter->description());
+                continue;
+            }
+
+            $command->addArgument(
+                $parameter->name(),
+                InputArgument::OPTIONAL,
+                $parameter->description(),
+                $parameter->defaultValue()
+            );
+        }
+
+        $command->assertCallable($task->code());
+        $command->idxTaskCode = $task->code();
+
+        return $command;
+    }
+
+    /**
+     * @param \ReflectionParameter $parameter
+     * @param string $description
+     */
+    private function addParameter(\ReflectionParameter $parameter, $description)
+    {
+        $name = $parameter->getName();
+
+        if (!$parameter->isOptional()) {
+            $this->addArgument($name, InputArgument::REQUIRED, $description);
+
+            return;
+        }
+
+        if ($this->isFlagOption($parameter)) {
+            $this->addOption($name, null, InputOption::VALUE_NONE, $description);
+
+            return;
+        }
+
+        $default = $parameter->getDefaultValue();
+        $this->addArgument($name, InputArgument::OPTIONAL, $description, $default);
     }
 
+
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $input = $this->filterByOriginalDefinition(
-            $input,
-            $this->getApplication()->getDefinition()
+        $input = $this->inheritDefinitionFrom(
```

Comment:
lo scopo di `inheritDefinitionFrom` è di eliminare eventuali parametri passati nella chiamata che non appartengono al task ma ad idephix stesso, corretto? In quel caso cercherei un nome del tipo `removeIdephixParamsFrom($input, $this->getApplication()->getDefinition())`ha senso?